C++ compile failure on addon with vrpc 2.4.0

**Describe the Bug**
Our project "profinet-stack" runs into a C++ compile error with vrpc-2.4.0 but was compiling fine with vrpc-2.3.2.

**Steps To Reproduce**

1. Unpack the attached zip archive that contains our simplified example for the project setup
2. In that project directory, call `npm install`
3. Watch how it fails with compile errors from `addon.cpp` in connection with `vrpc.hpp`

**Expected Behavior**
C++ compile with vrpc 2.4.0 should work just as it did with vrpc 2.3.2.

**Error message**
The compiler prints a lot of errors; the first one is this:

make: Entering directory '/home/foo/example-stack/build'
  CXX(target) Release/obj.target/cybus_profinet/node_modules/vrpc/vrpc/addon.o
In file included from ../node_modules/vrpc/vrpc/addon.cpp:46:
../node_modules/vrpc/vrpc/vrpc.hpp:1479:34: error: 'Ret' was not declared in this scope
 1479 |   _VRPC_MEMBER_FUNCTION_3(Klass, Ret, Function)
      |                                  ^~~
../node_modules/vrpc/vrpc/vrpc.hpp:1422:35: note: in definition of macro '_VRPC_MEMBER_FUNCTION_3'
 1422 |       Klass, decltype(static_cast<Ret (Klass::*)()>(&Klass::Function)),     \
      |                                   ^~~
../node_modules/vrpc/vrpc/vrpc.hpp:1278:19: note: in expansion of macro '_VRPC_VOID_MEMBER_FUNCTION_2'
 1278 | #define CAT(A, B) A##B
      |                   ^
../node_modules/vrpc/vrpc/vrpc.hpp:1279:27: note: in expansion of macro 'CAT'
 1279 | #define SELECT(NAME, NUM) CAT(_##NAME##_, NUM)
      |                           ^~~
../node_modules/vrpc/vrpc/vrpc.hpp:1287:30: note: in expansion of macro 'SELECT'
 1287 | #define VA_SELECT(NAME, ...) SELECT(NAME, VA_SIZE(__VA_ARGS__))(__VA_ARGS__)
      |                              ^~~~~~
../node_modules/vrpc/vrpc/vrpc.hpp:1305:3: note: in expansion of macro 'VA_SELECT'
 1305 |   VA_SELECT(VRPC_VOID_MEMBER_FUNCTION, __VA_ARGS__)
      |   ^~~~~~~~~
../cpp/binding.cpp:53:3: note: in expansion of macro 'VRPC_VOID_MEMBER_FUNCTION'
   53 |   VRPC_VOID_MEMBER_FUNCTION(ProfinetSniffer,
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~


**Desktop (please complete the following information):**

- OS: Linux, Fedora 34; gcc 11.1.1 (huh, quite high numbers these days)

[example-stack.zip](https://github.com/heisenware/vrpc/files/6580537/example-stack.zip)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   using `registerInstance`
 - missing class info messages when successfully reconnecting against a restarted
   broker
+- compiler error when using deprecated C++ binding macros on zero arguments
+  function
 
 ## [2.5.0] - 25 May 2021
 

--- a/vrpc/vrpc.hpp
+++ b/vrpc/vrpc.hpp
@@ -1476,7 +1476,7 @@ namespace vrpc {
           Ret>::registerAs(#Klass, #Function, FunctionDesc, {RetDesc});
 
 #define _VRPC_VOID_MEMBER_FUNCTION_2(Klass, Function) \
-  _VRPC_MEMBER_FUNCTION_3(Klass, Ret, Function)
+  _VRPC_MEMBER_FUNCTION_3(Klass, void, Function)
 
 #define _VRPC_VOID_MEMBER_FUNCTION_CONST_2(Klass, Function) \
   _VRPC_CONST_MEMBER_FUNCTION_3(Klass, void, Function)

